src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -0,0 +1,239 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static unit.GcsBucketControlled.listBucketResourceWithName;
+import static unit.GcsBucketControlled.listOneBucketResourceWithName;
+
+import bio.terra.cli.serialization.userfacing.resources.UFGcsBucket;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import com.google.cloud.storage.Bucket;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra resources` commands that handle referenced GCS buckets. */
+@Tag("unit")
+public class GcsBucketReferenced extends SingleWorkspaceUnit {
+
+  // external bucket to use for creating GCS bucket references in the workspace
+  private Bucket externalBucket;
+
+  @BeforeEach
+  @Override
+  protected void setupEachTime() throws IOException {
+    super.setupEachTime();
+    externalBucket = ExternalGCSBuckets.createBucket();
+    ExternalGCSBuckets.grantReadAccess(externalBucket, workspaceCreator.email);
+  }
+
+  @AfterEach
+  protected void cleanupEachTime() throws IOException {
+    ExternalGCSBuckets.deleteBucket(externalBucket);
```

Comment:
This test can leak buckets if there's a temporary error trying to delete the bucket, or if the test is interrupted during the middle of execution. CRL + Janitor is exactly for this use case. Consider using the `StorageCow` with the CleanupClient config set to talk to Janitor, and the tools Janitor is given permissions to delete buckets in this google project.

Reply:
Good idea. I'll add this to the follow-on ticket to replace this test project with a Terraformed one.

Two questions:
- Is there any difference in telling Janitor to clean up buckets in a project and running a script to wipe all buckets in the project say once a night? Just trying to understand what Janitor gives us here.
- I thought Janitor was being used for project cleanup, which are not as easily contained as buckets in a project. Are there any other places we're using Janitor to cleanup things other than projects?

Reply:
- You don't have to make sure your script runs successfully and it's called on all the relevant projects and those projects are allowed to have buckets that shouldn't be deleted.
- Yes, for the CRL-tests themselves where there is a terraformed gcp project and resources including buckets are created within them. It's pretty identical to what's going on here.